### DISCUSSION TITLE
Clarification on external extension setup in demo

### DISCUSSION THREAD

**Discussion by snoop244** [CodinGame/monaco-vscode-api#175] [General]
Hi,
I'm starting with your demo code and tried following the instructions to pull in a couple vsix files downloaded from the marketplace and dropped into the demo project and then imported after the vite config included the reference to `vsixPlugin() ` in the plugins section. 

Next, with the actual .vsix files in my project, I simply add them as imports at the top of your existing main.ts file. One provides language support for Python and the other is for docker. 

Am I importing them correctly? 
```
import '../ms-python.python-2023.15.12361009.vsix'
import '../ms-azuretools.vscode-docker-1.26.0.vsix'
```
When I start the application, it does not complain, but it also doesn't seem to load the extensions.

One other thing, I do notice when I start the application it does immediately complain about Typescript language features. I assume this is unrelated to the above issue, but I'm wondering if I am starting the application incorrectly. Is there some pre-processing step I need to invoke. Here is the Typescript language features errror from the terminal: `[vite] Internal server error: ENOENT: no such file or directory, open '/home/snoop244/Koriographer2/Koriographer/node_modules/vscode/dist/default-extensions/typescript-language-features/tsserver.web.js?v=6c505998' `

**Comment by CGNonofr** [comment]
Hello!

You seem to do everything correctly.

You'll only be able to load web compatible extensions though. If you want to know, go to vscode.dev and try to install an extension. If it works there, it should work here.

Unfortunately, a lot of extensions are not compatible.

It may change if we manage to make the remote extHost operational here but it'll require a server.

Regarding the ts erreur, I don't know, does the file exist?

**Reply by snoop244** [reply]
Yes, the tsserver.web.js file is there. 

WRT the extensions - I went to vscode.dev and, sure enough, the extensions I wanted are not supported. In the case of the Microsoft Python extension, it is only partly supported. 

My goal in exploring your framework was to see if I could introduce my own visual modelling tool as an editor, which is used for code generation, and then have LSP-based language support across a number of common languages (Python, C#,  Java, Typescript, Go and Rust). 

I had previously built the visual modelling tool as a VSIX a few years back. I found that process to be quite painful (message-passing, etc.) and was thinking your framework offered a path that was easier and more maintainable.

I'd appreciate your advice: if LSPs are my main goal + Manaco Editor + my own visual modeling tool, am I better off exploring Manaco Language Client?

Thx,
S.

**Reply by CGNonofr** [reply]
Yeah, probably!

The point of running vscode extension is to be able to use existing extensions. If you only want to create you app integrating a monaco editor, you'll have way more flexibility not doing it as a vscode extension.
